tests/testthat/test-unnecessary_concatenation_linter.R

```diff
@@ -45,18 +45,26 @@ test_that("unnecessary_concatenation_linter blocks disallowed usages", {
   )
 })
 
-test_that("Correctly handles concatenation within magrittr pipes", {
+local({
+  pipes <- c("%>%", "%!>%", "%T>%", "%<>%")
```

Comment:
Does the backpipe support strings as symbols?

`a <- 1; "a" %<>% c("b")` does what?

Reply:
Quite surprisingly it does the same thing as symbol `a` would do!

```r
a <- 1; "a" %<>% c("b")
a
# [1] "a" "b"
```

Reply:
Wow that's surprising behavior to say the least 🙈
I wouldv'e at least expected `a <- c("1", "b")`...

Reply:
You're right, it's even more surprising than my first understanding!

I guess it gets evaluated as

```r
"a" <- c("a", "b")
```

which fits the pattern of

```r
<expr> %<>% foo(y)
# equivalent to
<expr> <- foo(<expr>, y)
```

A great example of why `%<>%` should be avoided...

Reply:
Maybe a feature for a pipe linter to warn against using `%<>%` with non-SYMBOL LHS?

Reply:
For now I'm content with `%<>%` being covered by `assignment_linter()` to dissuade using it at all. Seems the simplest option.